common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -43,13 +43,13 @@
     private static final InternalLogger logger = InternalLoggerFactory.getInstance(NativeLibraryLoader.class);
 
     private static final String NATIVE_RESOURCE_HOME = "META-INF/native/";
-    private static final String OSNAME;
+    private static final boolean OSX;
     private static final File WORKDIR;
     private static final boolean DELETE_NATIVE_LIB_AFTER_LOADING;
 
     static {
-        OSNAME = SystemPropertyUtil.get("os.name", "").toLowerCase(Locale.US).replaceAll("[^a-z0-9]+", "");
-
+        String osname = SystemPropertyUtil.get("os.name", "").toLowerCase(Locale.US);
+        OSX = osname.startsWith("macosx") || osname.startsWith("osx");
```

Comment:
sure we can do but as there were no other places where this is used I did not bother.